internal/pkg/template/repository/manager/manager_test.go

```diff
@@ -52,7 +52,7 @@ func TestNew(t *testing.T) {
 	assert.NoError(t, err)
 	defer unlockFn()
 
-	assert.True(t, repo.Fs().Exists("example-file.txt"))
```

Comment:
není ten assert divný? Chápu, že je tam template1 ten přidals v internal/pkg/template/repository/manager/test/repository/.keboola/repository.json
Ale `example-file.txt` tam furt je.
A pokud je to teda tak jak předpokládám tak by tam mělo byt něco jako notExists `deprecated-template`.

Reply:
@zajca ten subor tam uz nie je, upravil som to aby sa to viac podobalo realite:
- testy, ktore pouzivaju git repo, maju uz rovnake git repo
- v `main` branch je `template1` + `deprecated-template`
- v `b1` je `template2` + `deprecated-template`.
- na to mame tento podrobny test: https://github.com/keboola/keboola-as-code/blob/michaljurecko-PSGO-255-fix-tmpl-preload/internal/pkg/template/repository/fs/gitfs_test.go
- a testuje sa tam presne stav `main` branche a `b1` branch

![image](https://github.com/keboola/keboola-as-code/assets/19371734/deb1c007-1efc-4f63-82d8-282d1a008414)

-----------------------

> A pokud je to teda tak jak předpokládám tak by tam mělo byt něco jako notExists deprecated-template.

Deprecated sablona nemoze mat zadanu path, takze takyto assert nedava zmyslel.
Tj to `continue` v kode nesposobi, ze sa dany adresar nestiahne z `git`, kedze deprecated sablona nema ziadny adresar. Sposobi iba to, ze sa deprecated sablonu nesnazime nacachovat do memory, pri starte API node. Ked aj tak `load` operacie neprejde, je pre deprecated sablony zakazana (zaujimaju nas iba metadata z `.keboola/repository.json`.

-------------------
Zmyslom toho assertu na ktory upozornuje je, ze sa "nieco" stiahlo, ze ten `fs` nie je prazdny. Nepotrebujem tam kontrolovat vsetky subory co tam su.

-------------------

Mozem pridat nejake dalsie asserts do ineho testu, do `TestRepositoryUpdate`, aj ked som to doteraz nepovazoval za potrebne, ... bude sa tym testovat, ze `git` cmd, ktory to interne spusta, spravil to co mal spravit ak neskonci chybou.

https://github.com/keboola/keboola-as-code/blob/3823163d86132c9b537fa97900b0f9b12209ad1d/internal/pkg/template/repository/manager/manager_test.go#L92

Reply:
Nie je to trivialna vec, takze chapem, ze to nechapes :) mal som to asi lepsie vysvetlit v komentaroch

Reply:
Tu som pridal asserty to toho ineho testu:
https://github.com/keboola/keboola-as-code/pull/1390/commits/9203f11b868e3108df1da458c965f3418e41055f